client/src/components/RequestDetailModal/CodeBlock.tsx

```diff
@@ -1,19 +1,28 @@
 import { Container, Paper } from '@mui/material';
 import React, { FC } from 'react';
 import useStyles from './styles';
+import { RequestHeader } from 'models/testSuiteModels';
+
+import { formatBodyIfJSON } from './helpers';
 
 export interface CodeBlockProps {
   body?: string | null;
+  headers?: RequestHeader[] | null | undefined;
 }
 
-const CodeBlock: FC<CodeBlockProps> = ({ body }) => {
+const CodeBlock: FC<CodeBlockProps> = ({ body, headers }) => {
   const styles = useStyles();
 
   if (body && body.length > 0) {
     return (
-      <Container component={Paper} variant="outlined" className={styles.codeblock}>
-        <pre>
-          <code>{body}</code>
+      <Container
+        component={Paper}
+        variant="outlined"
+        className={styles.codeblock}
+        data-testid="code-block"
+      >
+        <pre data-testid="pre">
+          <code data-testid="code">{body && formatBodyIfJSON(body, headers)}</code>
```

Comment:
```suggestion
          <code data-testid="code">{formatBodyIfJSON(body, headers)}</code>
```
`body` was already checked in line 16.

Reply:
Good catch, the linter must have led me to this at one point.